### DISCUSSION TITLE
Moving from hardhat-based cousre to this course

### DISCUSSION THREAD

**Discussion by emtothed** [Cyfrin/foundry-full-course-cu#609] [Q&A]
Hello everyone

I've been watching the other course (hardhat-based one) and I just found out about this course and I want to start watching this one now.

My first question is which one is better to learn, hardhat or foundry ? i've done my research but there is no certain answer.
I konw solidity basics now , and i'm also familiar with js but not that much, and working with solidity is easier for me and i like it but i want to know which one is actually better in work

And my second question is : 
I watched the other course until the end of the **remix fund me** 
Are that parts that I've watched (blockchain basics and remix part )totally the same with the lessons in this course? isn't there any differences or updates ?
Is it ok if I start watching this course from Lesson 5 : AI prompting ?

**Comment by theirrationalone** [comment]
Answer 1: [Read This](https://github.com/Cyfrin/foundry-full-course-f23/discussions/522#discussioncomment-6620940)

Answer 2: Yes, Absolutely you can, You can start this course from Lesson 5, But I will say, Please at least watch that Lesson 0 === Introduction, of this new course.

**Reply by emtothed** [reply]
thank you so much for your help @theirrationalone

**Reply by theirrationalone** [reply]
and one more thing, Only Chapter 12 will be challenging to you (a bit only), I'll suggest you to, Please Jump to Lesson 12 of this new course and watch until contract's implementation only and concrete your understanding on **What's happening into that contract** but don't go further. and then again jump back to Lesson 5 and start watching the course to understand foundry's tooling. It will save your Precious time. I'm suggesting you this because as You said that you completed Patrick's HH course.

# You only need to Understand Foundry's Tooling...!

**Reply by emtothed** [reply]
hello again @theirrationalone 

I didn't complete the js course i just watched it to the end of **Lesson 4: Remix Fund Me**

and I read your answer on the other page but I can't still decide which course should I watch.
I can not find out if hardhat is better for me to learn or the foundry. I did some search and hardhat is the most popular one based on articles but people are talking about how good foundry is in forums and I don't know enough to understand which one is better for me.
and about the courses , i cant decide about them too actually.

I think gaining full-stack knowledge and knowing about next js and front-end can be something really beneficial because i don't know enough about it but I don't know if it would be really useful or not. i feel like i'm missing it if i don't watch the js course and i don't know if it's a good thing to do to watch both courses. 

so i'm really confused right now and it would be a huge help if you guide me more if you can.

thank you

**Reply by emtothed** [reply]
Can I watch the foundry course and learn the web3 and AI concepts from it and then go for java script course if i wanted and ignore the hardhat concepts and take the full stack points from it and implement it with foundry ? does foundry have the full-stack features that hardhat have and is it better than hardhat ?

**Reply by theirrationalone** [reply]
Hey @emtothed 

Learning Fullstack is Still Worth a lot & a Great Choice too. 

You can upgrade your skills on frontend & backend by diving more into js, reactJs, node(node with expressJs), nextJs.

# If you wanna be a Fullstack only... 

- Js (You can skip DOM) 

- ReactJs (Learn it from your heart, conquer it) 

- NodeJs with expressJs(expressJs is most powerful in building REST APIs). 

- NEXTJS (easiest after learning react and node) 


And as you said you're familiar with Js, So you can learn ReactJs in depth. 

And from NodeJs You can learn only file handling & REST API's with Or without expressJs.


You can learn CSS basics And then shift to tailwind css. 

# If you wanna be a Blockchain Expert. 

After learning Fullstack, it will a Plus ➕ in your Blockchain Journey, You will be able to play in both playgrounds, Blockchain WEB3 & Blockchain Security expert(if you take this course and upcomming one) 


My recommendations are... 

As you now acquired lots of knowledge about blockchain & WEB3... So it would be great choice to take this new foundry's course. 


You have 4 Great Choices to Conquer on... 

1. ReactJs
2. Blockchain
3. NodeJs. 
4.Solidity

And Please don't feel scared to take this course... This course is all about tooling And foundry is just a tool. 

If you have acquired 100% of blockchain & solidity part of Patrick's HH course. Then this will be a cake bite to you. And please don't feel worried about moralis & graph, they're just helpers to structure WEB3 And to monitor & store event logs data on blockchain. After learning ReactJs it will be a easy Game to implement moralis And graph in your DAPPS. 

# Lets conclude all... 

What you have learned. 

- Blockchain & solidity Basics ✅
- Js ✅
- Hardhat ✅

What remains to learn. 

- ReactJs in Depth (recommended course: Academind maximilian schwarzmuler Paid but most effective) 

- NodeJs (Basics + file handling + RESTFull APIs) (mdn docs) 

- Blockchain & Solidity Advanced. (This course) 

- Blockchain Security course (Patrick will release it soon) 


Hardhat is similar to Foundry but HH is there to support fullstack and Foundry is to support security of smart contracts. 

And please read my last reply also.

**Reply by emtothed** [reply]
dear @theirrationalone 

first of all thank you a lot for your complete explanation , actually i don't know how to thank you for your effort .

i think i made you misunderstand. excuse me for that
first i have to correct you : i don't know hardhat i've just learned solidity basics through the js/web3 course .

and please let me specify my goal . my goal is to be a smart contract developer through learning solidity and how to implement smart contracts with it. and my focus is not on the front-end , it is mostly on building the smart contracts and make them secure and generally the blockchain back-end job. but i think knowing about the front-end can be beneficial . i think learning the front-end parts of the  js/solidity course of mr. collins can help me to understand how the things are getting done in dapplications and also makes me be able to make a front-end for my projects so they will be represented better in the future in my resumes. but my goal is not to be a full-stack developer or front-end developer . i don't want to go in dept about react and front-end development ,just want to know a little bit .and i think mr.collins js/solidity course can give it to me. but this new course is offering a better fremwork and is more updated and also has AI related concepts . but i don't know if the hardhat is a better framework or the foundry .

if i want to learn the foundry , in your opinion, can i watch the new course and then learn full-stack stuff from the js/solidity course and implement it with foundry or it can not be done ?

**Reply by theirrationalone** [reply]
Ohhhhhh Sorry  @emtothed  for misunderstanding you 😅

So according to your Goal, This course suits best to you and let me clear one thing, In context to keep smart contracts secure and decentralized, there is no need to understand REACTJS, NODEJS, NEXTJS, absolutely nothing, Fullstack has no role in making & securing Smart contracts and blockchain.

But knowing little bit about all of them Js's frameworks is a Plus btw, If you only have a bit of knowledge about RESTFUL APIs, Js, REACTJs, then you can understand what's going into a DAPP WEB3 application.

So take this course and WELCOME to Blockchain (Foundry).

**Reply by emtothed** [reply]
dear @theirrationalone I really appreciate your help and your valuable time thank you a lot for everything .
